ksqldb-functional-tests/src/main/java/io/confluent/ksql/test/driver/AssertExecutor.java

```diff
@@ -74,28 +74,35 @@
           CommonCreateConfigs.KAFKA_TOPIC_NAME_PROPERTY
       )).add(new SourceProperty(
           ds -> ds.getKsqlTopic().getKeyFormat().getFormatInfo().getFormat(),
-          (cs, cfg) -> cs.getProperties().getKeyFormat(cs.getName()).map(FormatInfo::getFormat)
+          (cs, cfg) -> cs.getProperties().getKeyFormat(cs.getName(), cfg).map(FormatInfo::getFormat)
               .orElse(cfg.getString(KsqlConfig.KSQL_DEFAULT_KEY_FORMAT_CONFIG)),
           "key format",
           CommonCreateConfigs.KEY_FORMAT_PROPERTY,
           CommonCreateConfigs.FORMAT_PROPERTY
       )).add(new SourceProperty(
-          ds -> ds.getKsqlTopic().getKeyFormat().getFormatInfo().getFormat(),
-          (cs, cfg) -> cs.getProperties().getKeyFormat(cs.getName()).map(FormatInfo::getFormat)
-              .orElse(cfg.getString(KsqlConfig.KSQL_DEFAULT_KEY_FORMAT_CONFIG)),
+          ds -> ds.getKsqlTopic().getKeyFormat().getFormatInfo().getProperties(),
+          (cs, cfg) -> cs.getProperties().getKeyFormatProperties(
```

Comment:
This bugfix is unrelated to the changes in my PR (but I happened to notice while in here updating function signatures): previously this check was checking for matching key formats (a duplicate of the above check) even though the name suggests that it checks for key format properties. I've fixed it to properly check for key format properties instead of key formats.